### PR TITLE
Upgrade DRF version.

### PR DESCRIPTION
**Pull request by doctoryes**
https://openedx.atlassian.net/browse/PLAT-1604

As part of an upgrade of django-rest-framework for edx-platform, this dependency needs to have its DRF version upgraded as well. Upgraded the DRF version, fixed tests, and bumped the version number.

### CODE REVIEW

File: edxval/serializers.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def to_internal_value(self, data):
         Convert data into CourseVideo instance and image filename tuple.
         """
         course_id = data
-        course_video = image = None
+        course_video = image = ''
```
Comment (muhammad-ammar) [inline edxval/serializers.py]: Any particular reason for empty string instead of `None`?

File: requirements.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 django>=1.8,<1.9
-djangorestframework>=3.1,<3.2
+git+https://github.com/edx/django-rest-framework.git@1ceda7c086fddffd1c440cc86856441bbf0bd9cb#egg=djangorestframework==3.6.3
```
Comment (mushtaqak) [inline requirements.txt]: Why are we using this drf commit? If there is any necessity then is there any plan to upgrade to stable release than one particular commit ?

File: edxval/tests/test_views.py [hunk 3]
```diff
@@ -498,7 +498,7 @@ def test_post_videos(self):
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
         self.assertEqual(
             response.data.get("edx_video_id")[0],
-            "This field must be unique."
+            u'video with this edx video id already exists.'
```
Comment (mushtaqak) [inline edxval/tests/test_views.py]: @doctoryes why did we need to change this ?

File: edxval/serializers.py (hunk 1 above)
Comment (doctoryes) [inline edxval/serializers.py]: The newest version of DRF (v3.6.3) is apparently stricter about empty strings vs. `None`. Defaulting to `None` and passing `None` through the DRF machinery caused a few tests to fail in edx-platform, while using an empty string fixed the tests.

File: requirements.txt (hunk 2 above)
Comment (doctoryes) [inline requirements.txt]: The reason why is documented here:
https://openedx.atlassian.net/browse/PLAT-1581?focusedCommentId=270321&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-270321
It'd be *great* to use the stable release again - perhaps we can get to that point soon.

File: edxval/tests/test_views.py (hunk 3 above)
Comment (doctoryes) [inline edxval/tests/test_views.py]: This test raises a DRF `ValidationError` and the message associated with that exception was changed to let the Django-generated message be returned. So I changed the test to check for the new more-specific message.

Comment (doctoryes) [inline edxval/tests/test_views.py]: The code:
https://github.com/encode/django-rest-framework/blob/3.6.3/rest_framework/exceptions.py#L121-L140

**Pull request merged**